### PR TITLE
Add swift-nightly

### PR DESCRIPTION
**Pull request by deevus**
This adds a nightly version for Swift

I copied the manifest from the main bucket and updated it to get the nightly version from the official download page. 

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by deevus** [comment]
This is not quite working. It seems as though the archive used for the development snapshots are not the same as release. 

```
scoop install .\bucket\swift5.10-nightly.json
WARN  Purging previous failed installation of swift5.10-nightly.
ERROR 'swift5.10-nightly' isn't installed correctly.
Removing older version (5.10-DEVELOPMENT-SNAPSHOT-2024-02-27).
'swift5.10-nightly' was uninstalled.
Installing 'swift5.10-nightly' (5.10-DEVELOPMENT-SNAPSHOT-2024-02-27) [64bit] from 'C:\Users\Simon\Projects\ScoopVersions\bucket\swift5.10-nightly.json'
Loading swift-5.10-DEVELOPMENT-SNAPSHOT-2024-02-27-a-windows10.exe from cache.
Checking hash of swift-5.10-DEVELOPMENT-SNAPSHOT-2024-02-27-a-windows10.exe ... ok.
Running pre_install script...
ERROR Exit code was 1620!
Failed to extract files from C:\Users\Simon\scoop\apps\swift5.10-nightly\5.10-DEVELOPMENT-SNAPSHOT-2024-02-27\extract_folder\AttachedContainer\a0.
Log file:
  ~\scoop\apps\swift5.10-nightly\5.10-DEVELOPMENT-SNAPSHOT-2024-02-27\extract_folder\AttachedContainer\msi.log
```

The mentioned log file is empty. 

@FlawlessCasual17 Pinging you since you created the original Swift manifest and have been submitting updates for it

**Comment by deevus** [comment]
This is using nightly builds from The Browser Company, since they are currently not offered officially by Apple

File: bucket/swift-nightly.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        "Remove-Item \"$dir\\Swift\\Toolchains\\$ver+Asserts\" | Out-Null",
+        "# Remove-Item \"$dir\\Swift\\Tools\\$ver\" | Out-Null",
+        "Remove-Item \"$dir\\extract_folder\" -Recurse | Out-Null",
+        "Move-Item -Path \"$dir\\Swift\\*\" -Destination \"$dir\"",
+        "Remove-Item \"$dir\\Swift\" | Out-Null"
+    ],
+    "env_add_path": [
+        "Runtimes\\usr\\bin",
+        "Toolchains\\usr\\bin",
+        "Tools"
```
Comment (tech189) [inline bucket/swift-nightly.json:52]: Is it okay to leave `Tools` in the path if it doesn't exist?

Comment (deevus) [reply bucket/swift-nightly.json:52]: Yeah it works fine. Might be worth deleting though

**Comment by FlawlessCasual17** [comment]
Hey, do you guys still need help, sorry for not responding. I was busy with IRL stuff, like university.

**Comment by deevus** [comment]
> @deevus Hey, do you guys still need help, sorry for not responding. I was busy with IRL stuff, like university.

No problem. All good now.

**Pull request merged**